drivers/sensors/l86xxx_uorb.c

```diff
@@ -0,0 +1,754 @@
+                  if (frame.sats[i] != 0)
+                    {
+                      ++sats;
+                    }
+                }
+
+              gps.satellites_used = sats;
+            }
+          break;
+        }
+
+        /* GSV and GLL data are transmitted by the l86-XXX but do not provide
+        additional information. Since GLL is always the last message
+        transmitted, events will be pushed whenever that sentence is read
+        */
+
+        case MINMEA_SENTENCE_GLL:
+        {
+            dev->lower.push_event(dev->lower.priv, &gps, sizeof(gps));
+        }
+
+        /* All remaining sentences are not transmitted by the module */
+
+        case MINMEA_SENTENCE_GSV:
+        case MINMEA_SENTENCE_GBS:
+        case MINMEA_SENTENCE_GST:
+        case MINMEA_SENTENCE_ZDA:
+        {
+          break;
+        }
+
+        case MINMEA_INVALID:
+        {
+          break;
+        }
+
+        case MINMEA_UNKNOWN:
+        {
+          break;
+        }
+      }
+
+      nxmutex_unlock(&dev->devlock);
+    }
+
+  return 0;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: l86xxx_register
+ *
+ * Description:
+ *   Register the L86-XXX GNSS device driver.
+ *
+ * Arguments:
+ *    devpath   -  The device path to use for the driver
+ *    uartpath  -  The path to the UART character driver connected to the
+ *                 transceiver
+ *    devno     -  The device number to use for the topic (i.e. /dev/mag0)
+ ****************************************************************************/
+
+int l86xxx_register(FAR const char *devpath,
+              FAR const char *uartpath, int devno)
+{
+  FAR l86xxx_dev_s *priv = NULL;
+  int err;
+
+  DEBUGASSERT(devpath != NULL);
+  DEBUGASSERT(uartpath != NULL);
+
+  /* Initialize device structure */
+
+  priv = kmm_zalloc(sizeof(l86xxx_dev_s));
+  if (priv == NULL)
+    {
+      snerr("Failed to allocate instance of L86-XXX driver.\n");
+      return -ENOMEM;
+    }
+
+  memset(priv, 0, sizeof(l86xxx_dev_s));
+
+  /* Initialize mutex */
+
+  err = nxmutex_init(&priv->devlock);
+  if (err < 0)
+    {
+      snerr("Failed to initialize mutex for L86-XXX device: %d\n", err);
+      goto free_mem;
+    }
+
+  /* Initialize semaphore */
+
+  err = nxsem_init(&priv->run, 0, 0);
+  if (err < 0)
+    {
+      snerr("Failed to register nau7802 driver: %d\n", err);
+      goto destroy_mutex;
+    }
+
+  /* Open UART interface for use */
+
+  err = file_open(&priv->uart, uartpath, O_RDWR | O_CLOEXEC);
+  if (err < 0)
+    {
+      wlerr("Failed to open UART interface %s for L86-XXX driver: %d\n",
+            uartpath, err);
+      goto destroy_sem;
+    }
+
+  /* Setup sensor with configured settings */
+
+  read_line(priv); /* Wait until module is powered on */
```

Comment:
Is this necessary? Does the module output a boot-up message? Maybe you can instead perform a `file_poll` operation to wait for some messages to be ready in the UART buffer. What does the L86 datasheet recommend doing to ensure the device is on?

Reply:
Replacing with a check for startup as described by the data sheet